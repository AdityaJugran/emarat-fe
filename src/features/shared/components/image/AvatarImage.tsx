import { Image } from 'antd';
import { transformCloudinaryImage } from 'lib/utils';
import userPlaceholderImg from 'assets/images/user-placeholder.svg';

export default function AvatarImage({
	userImg,
	size,
}: {
	userImg: string;
	size?: number;
}) {
	return (
		<Image
			className="rounded-full"
			width={size || 40}
			height={size || 40}
			preview={false}
			src={
				transformCloudinaryImage(`${userImg}`, 'AVATAR') ||
				userPlaceholderImg
			}
			fallback={userPlaceholderImg}
			alt="avatar"
		/>
	);
}
