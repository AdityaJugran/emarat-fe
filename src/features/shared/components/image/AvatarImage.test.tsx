import AvatarImage from 'features/shared/components/image/AvatarImage';
import renderer from 'react-test-renderer';

it('Image Template', () => {
	const userImg =
		'https://res.cloudinary.com/emarat/image/upload/v1631362403/dfejes8g9doeysyniydd.png';
	const tree = renderer.create(<AvatarImage userImg={userImg} />).toJSON();
	expect(tree).toMatchSnapshot();
});
