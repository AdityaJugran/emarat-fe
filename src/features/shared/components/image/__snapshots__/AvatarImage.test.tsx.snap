// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Template 1`] = `
<div
  className="ant-image"
  style={
    Object {
      "height": 40,
      "width": 40,
    }
  }
>
  <img
    alt="avatar"
    className="ant-image-img rounded-full"
    onError={[Function]}
    onLoad={[Function]}
    src="https://res.cloudinary.com/emarat/image/upload/w_50,h_50,c_thumb,g_face,r_max/v1631362403/dfejes8g9doeysyniydd.png"
    style={
      Object {
        "height": 40,
      }
    }
  />
</div>
`;
